### PR TITLE
feat: Recordings in notebooks

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

A range of WIP work to improve the support of things in Notebooks

## Changes

* Adds the ability to drag and drop items in (although the UX here isn't super clear _yet_) 
* Supports (in a preview state) Persons, Flags, Recordings, Playlists
* Some style tweaks
* Placeholder for adding images
* Added a "personLogic" for individual person loading
* Load recording in paused state when in a notebook
* Removed an old optimisation for up next recordings that broke some other things
* Adds special playlist view for going back and forth from the playlist to an item in notebooks
* Fixed a bug with playlist updating params even if the setting was off

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 1]
```diff
@@ -190,17 +190,29 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
             },
         ],
     })),
-    listeners(({ values, props, actions, cache }) => ({
-        loadEntireRecording: () => {
-            actions.loadRecordingMeta()
-        },
-        loadRecordingMetaSuccess: () => {
+    listeners(({ values, actions, cache, props }) => ({
+        loadRecording: ({ full }) => {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:194]: so `!full` means `only load metadata` ? 

That threw me... Should it be something like `previewOnly` 

Not blocking but 🤷

**Review by pauldambra — COMMENTED** [review]
# weird interaction between modal player and adding recording to notebook

If I clicked add to notebook on a recording it would open the modal player, then clicking add to notebook on the modal player would add the recording to the notebook *but* behind the fullscreen blur

Then any subsequent autoplay of "next recording" would launch the modal.

I'll slack a GIF since it's a 35MB gif 🤣 

## Adding persons

I managed to get in a weird state, as well as struggling to drag and drop the person. But then I couldn't load anything because the notebook couldn't load the person. So, we probably need a new error boundary to stop notebooks killing the page

File: frontend/src/scenes/notebooks/Notebook/Notebook.tsx [hunk 2]
```diff
@@ -32,13 +33,47 @@ export function Notebook({ id, sourceMode, editable = false }: NotebookProps): J
...
+                        view.pasteText(text)
+
+                        return true
+                    }
+
+                    if (event.dataTransfer.files && event.dataTransfer.files[0]) {
+                        // if dropping external files
+                        const file = event.dataTransfer.files[0] // the dropped file
+
+                        console.log('TODO: Dropped file!', file)
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/Notebook.tsx:68]: This ones on purpose

**Comment by benjackwhite** [comment]
@pauldambra looks like there's a naughty conditional hook in the PropertiesTable. Will fix that up... These notebooks are finding all sorts of existing issues elsewhere 😅 [reminds me of...](https://www.youtube.com/watch?v=AbSehcT19u0)

**Pull request merged**